middle_end/flambda/types/env/aliases.ml

```diff
@@ -982,7 +984,7 @@ let get_canonical_element_exn t element elt_name_mode ~min_name_mode
   let canonical_element, name_mode, coercion_from_canonical_to_element =
     match canonical t element with
     | Is_canonical ->
-      element, elt_name_mode, Coercion.id
+      Simple.without_coercion element, elt_name_mode, Simple.coercion element
```

Comment:
Sorry, I am having troubles paging in my knowledge about aliases;
is this change for the sake of readability, or do you expect different
values? (It looks like there are other places were we assume that
`Is_canonical` implies no/identity coercion.)

Reply:
I was also unclear what the invariant was supposed to be about coercions on canonical aliases.  Can we add some assertions to make this clear?

Reply:
Yes, it's definitely intended that a canonical element has no coercion. Adding assertions is a good idea (and they'll be cheap because it's just a flag check on the integer id).

Reply:
(Indeed, and if they were expensive we could still have them
guarded by a command-line switch.)